schema.cc

```diff
@@ -101,11 +102,21 @@ std::ostream& operator<<(std::ostream& os, ordinal_column_id id)
     return os << static_cast<column_count_type>(id);
 }
 
+std::unique_ptr<dht::i_partitioner> default_partitioner;
+
+void schema::set_default_partitioner(const sstring& class_name, unsigned ignore_msb)
+{
+    default_partitioner = dht::make_partitioner(class_name, smp::count, ignore_msb);
+}
+
 const dht::i_partitioner& schema::get_partitioner() const {
     if (_raw._partitioner) {
         return *_raw._partitioner;
     }
-    return dht::global_partitioner();
+    if (!default_partitioner) {
+        default_partitioner = std::make_unique<dht::murmur3_partitioner>(smp::count, 12);
+    }
+    return *default_partitioner;
```

Comment:
Why is all this done in the getter? Can't we do this in the constructor and make sure `_raw._partitioner` is never null, and points to `default_partitioner` instead?

Reply:
This patch just moves code from one place to the other. Following patches change this.